NZSLDict/Classes/SignsDictionary.m

```diff
@@ -145,27 +170,30 @@ - (DictEntry *)findExact:(NSString *)target
 
 - (NSArray *)searchFor:(NSString *)target
 {
-    NSMutableArray *sr = [NSMutableArray array];
+    NSMutableOrderedSet *sr = [[NSMutableOrderedSet alloc] init];
     NSString *exactTerm = normalise(target);
     NSString *containsTerm = [NSString stringWithFormat:@"%%%@%%", exactTerm];
+    NSString *startsWithTerm = [NSString stringWithFormat:@"%@%%", exactTerm];
     
     sqlite3_stmt *exactPrimaryMatchStmt;
+    sqlite3_stmt *startsWithPrimaryMatchStmt;
     sqlite3_stmt *containsPrimaryMatchStmt;
     sqlite3_stmt *exactSecondaryMatchStmt;
     sqlite3_stmt *containsSecondaryMatchStmt;
     
     bool statementPreparedOk = true;
-    statementPreparedOk = sqlite3_prepare_v2(db, "SELECT * FROM words WHERE gloss = ? OR maori = ?", -1, &exactPrimaryMatchStmt, NULL) == SQLITE_OK &&\
+    statementPreparedOk = sqlite3_prepare_v2(db, "SELECT * FROM words WHERE lower(gloss) = ? OR lower(maori) = ?", -1, &exactPrimaryMatchStmt, NULL) == SQLITE_OK &&\
+                          sqlite3_prepare_v2(db, "SELECT * FROM words WHERE gloss LIKE ? OR maori LIKE ?", -1, &startsWithPrimaryMatchStmt, NULL) == SQLITE_OK &&\
```

Comment:
Do you not need something like the `%` that mimics the `LIKE "STRING%"` behaviour for a starts with search?